File: 50-powx-n/50-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0 : return 1
        if n < 0 : return 1 / self.myPow(x, -n)
        
        ret = 1 if n % 2 == 0 else x 
        return ret * self.myPow(x*x, n // 2)

File: 50-powx-n/50-powx-n.cpp
#include<cmath>
using namespace std;
class Solution {
public:
    double myPow(double x, int n) {
        if(n == 0) return 1;
        double ret = (abs(n)%2==1) ? x : 1;
        if( n < 0 ){
            return 1 / (ret * myPow(x*x, -(n/2) ));   
        }else{
            return 1 * ret * myPow(x*x, n/2);   
        }
        
    }
};